examples/verify_proof.rs

```diff
@@ -7,28 +7,65 @@ fn read_json<T: DeserializeOwned>(path: &str) -> T {
 }
 
 fn main() {
-    let block: AxonBlock = read_json("examples/block.json");
+    let block: AxonBlock = read_json("examples/block1.json");
+    println!("block: {:?}", block);
     let proof: Proof = read_json("examples/proof.json");
     let metadata: Metadata = read_json("examples/metadata.json");
     let mut validators = metadata
         .verifier_list
         .iter()
-        .map(|v| Validator {
+        .map(|v| ValidatorExtend {
             bls_pub_key:    v.bls_pub_key.clone(),
+            pub_key:        v.pub_key.clone(),
             address:        v.address,
             propose_weight: v.propose_weight,
             vote_weight:    v.vote_weight,
         })
         .collect::<Vec<_>>();
 
     let previous_state_root =
-        hex::decode("3ae76798c8eaaf3005455c254b7ca499b0de32cf5fdf0d42e967059806d93a37").unwrap();
+        hex::decode("0000000000000000000000000000000000000000000000000000000000000000").unwrap();
 
-    assert!(axon_tools::verify_proof(
+    let result = axon_tools::verify_proof(
         block,
         H256::from_slice(&previous_state_root),
         &mut validators,
         proof,
-    )
-    .is_ok());
+    );
+    println!("verify_proof: {:?}", result);
+
+    assert!(result.is_ok());
+}
+
+#[cfg(test)]
+mod tests {
+    use axon_tools::hash::keccak_256;
+    use axon_tools::types::{Proof, Proposal};
+    use ethers_core::utils::rlp::Encodable;
+
+    #[test]
+    fn test_proposal() {
+        let proposal = Proposal {
+            version:                  Default::default(),
+            prev_hash:                Default::default(),
+            proposer:                 Default::default(),
+            prev_state_root:          Default::default(),
+            transactions_root:        Default::default(),
+            signed_txs_hash:          Default::default(),
+            timestamp:                0,
+            number:                   100,
+            gas_limit:                Default::default(),
+            extra_data:               Default::default(),
+            base_fee_per_gas:         Default::default(),
+            proof:                    Proof::default(),
```

Comment:
The mocked data should be generated by `rand`.

Reply:
if rand is used, the final result won't be fixed, which can not be verified.
this test is used to compare the axon-tools generated proposal hash is the same with axon.